### PR TITLE
Simplify Message and Packets (+ adding NodeId)

### PR DESCRIPTION
**Pull request by matteo-dinoia**
In this PR:
1. Message type is complex. To simplify it I put request and response first and foremost (because server is only interested in request and send only response).
2. I also merged Text and Media Request Response (the enums)
3. added serverType (could be better)
4. added NodeId for readability
5. added example of use of Message Content

Edit: see comment below for newest changes

### CODE REVIEW

**Comment by Marui-9** [comment]
Makes sense 👍

**Comment by matteo-dinoia** [comment]
Set as priority because it create base for common code

**Comment by matteo-dinoia** [comment]
Update code to remove the high level of nesting in favor of even simpler type.
Testing the usability of the nested code,  I found that while it did add a lot of complexity during creation, it didn't simply almost anything in unpacking.

**Review by IlyaEmeliyanov — CHANGES_REQUESTED** [review]
Well done, **small change**: I think that for the **ServerType** it'd be cleaner to use an **enum** with 3 variants (Chat, Text and Media) instead of a struct with booleans.

**Comment by matteo-dinoia** [comment]
> Well done, **small change**: I think that for the **ServerType** it'd be cleaner to use an **enum** with 3 variants (Chat, Text and Media) instead of a struct with booleans.

Yes, but it wouldn't allow for server of multiple kind which the professor was in favor of. I think we can vote on this separately.

**Comment by Suge42** [comment]
The changes makes sense, but the simplification of the Message type is reverting a previous changed on which all WGL agreed, so we'll vote about that.
Other than that, the rest of the changes like Nack, those at packets and similar, and other small things looks fine.

File: AP-protocol.md [hunk 1]
```diff
@@ -157,141 +166,87 @@ Message is subject to fragmentation: see the dedicated section.
...
-}
-
-struct MediaRequest{
-	Media{  //C -> S : media?(media_id, +media type+ )
-		media_id: u64
-		///media: media_kind
-	}
+#[derive(Debug)]
+pub struct MessageData { // Only part fragmentized
+	source_id: NodeId,
```
Comment (luiss07) [inline AP-protocol.md:185]:
```suggestion
```
The `source_id` is already present into the `SourceRoutingHeader`.

Comment (matteo-dinoia) [reply AP-protocol.md:185]: Fair. I'll remove it now. i still think it is still may be better to keep it here to isolate different layer. But i'll open a PR about it.

File: AP-protocol.md [hunk 2]
```diff
@@ -157,141 +166,86 @@ Message is subject to fragmentation: see the dedicated section.
...
 ```rust
-struct Error {
-	session_id: u64,
-	/// ID of drone, server of client that is not a neighbor:
-	id_not_neighbor: String,
-	ttl: u32,
-}
+let routing = getRoutingHeader();
+let content = MessageType:ReqFile(8);
+Message::new(ruting, source_id, session_id, content)
```
Comment (brokenhouse) [inline AP-protocol.md:222]:
```suggestion
Message::new(routing, source_id, session_id, content)
```

File: AP-protocol.md [hunk 3]
```diff
@@ -157,141 +166,86 @@ Message is subject to fragmentation: see the dedicated section.
...
 
-This message cannot be dropped by drones due to Packet Drop Rate.
+This message cannot be dropped by drones due to Packet Drop Probability.
 
 ```rust
-struct Dropped {
-	session_id: u64,
+pub enum Nack{
+	ErrorInRouting(NodeId), // contains id of not neighbor
+	Dropped
```
Comment (chiarastrozzii) [inline AP-protocol.md:238]: i think it would be useful to keep track of which fragment we have dropped, just to know which one we have to send again

**Comment by Marui-9** [comment]
The `Message(Fragment)` naming is ambiguous as it could introduce a conflict between low level fragments and higher level Messages. It could be be beneficial to have something like `FragPack(Fragment)`.

**Pull request merged**